pkgs/by-name/ov/ovn/generic.nix

```diff
@@ -0,0 +1,92 @@
+{
+  version,
+  hash,
+  updateScriptArgs ? "",
+}:
+
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  autoreconfHook,
+  gnused,
+  libbpf,
+  libcap_ng,
+  numactl,
+  openssl,
+  pkg-config,
+  procps,
+  python3,
+  unbound,
+  xdp-tools,
+  writeScript,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ovn";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "ovn-org";
+    repo = "ovn";
+    rev = "refs/tags/v${version}";
+    inherit hash;
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+    python3
+  ];
+
+  buildInputs = [
+    libbpf
+    libcap_ng
+    numactl
+    openssl
+    unbound
+    xdp-tools
+  ];
+
+  # need to build the ovs submodule first
```

Comment:
Can we tell it to use our ovs builds, or is it something else?

Reply:
It needs a copy of the source and a build output. The installed output we generate from ovs is not sufficient for the latter. I’ve explored a few different options including using the ovs src or even creating an extra artifact from the ovs build. What sold me on using the submodule is issues matching the expected version. Even the LTS build assumes a much newer ovs than the ovs LTS, which surprised me. 

I can revisit if desired, but using the submodule will always ensure we are building it with the same ovs rev they are.

Reply:
I should note that this extra ovs built isn’t actually installed, it’s just used will building ovn.

Reply:
Now that I think about it, I’ll look into this again and make sure it’s not linking to these.

Reply:
I don't see any linking to ovs, so that shouldn't be a problem. I'm inclined to leave this as written since it's the upstream way and potentially less fragile.